day16/src/lib.rs

```diff
@@ -240,11 +238,10 @@ blueprint! {
         // Generate a random number
         // WARNING: DON'T USE THIS IN PRODUCTION !
         fn random_number(&mut self, min: i32, max: i32) -> usize {
-            let mut random_number: Decimal = (self.random_seed * (self.nonce as i32)).into();
-            random_number = (random_number.abs() / i32::MAX) * (max - min) + min;
-
-            self.nonce += 1;
-            random_number.to_string().split(".").collect::<Vec<&str>>().get(0).unwrap().parse().unwrap()
+            self.random_seed = ( ( 1664525 * self.random_seed ) + 1013904223 ) % 4294967296;
```

Comment:
Don't we need to keep the nonce ? I think that without it, it will always return the same number

Reply:
no because the `random_seed` is updated each time, it's on the right and left side of the equation

a Linear Congruential Generator does: `X =  ( (a * X) + c ) % d` where X is `random_seed`

Reply:
Ohhh, really cool